clients/src/main/java/org/apache/kafka/clients/producer/ProducerConfig.java

```diff
@@ -79,14 +79,15 @@
                                            + " durability of records that are sent. The following settings are allowed: "
                                            + " <ul>"
                                            + " <li><code>acks=0</code> If set to zero then the producer will not wait for any acknowledgment from the"
-                                           + " server at all. The record will be immediately added to the socket buffer and considered sent. No guarantee can be"
+                                           + " server at all. The record will be immediately added to the socket buffer and considered sent. There is no guarantee"
+                                           + " that the socket buffer will be flushed prior to the shutdown of the producer JVM.  No guarantee can be"
```

Comment:
@ijuma the problem I believe is in the semantics of what `complete` is for an acks=0 request.  Going through `Sender.run()`, we call `send` then immediately start `poll` for responses.   Inside of poll we end up in [handleCompletedSends](https://github.com/cotedm/kafka/blob/KAFKA-3129/clients/src/main/java/org/apache/kafka/clients/NetworkClient.java#L428) which is actually just faking a response.  Then with the fake response in hand, we just immediately handle the callback we got.  We're not *actually* waiting for the request to go down to the OS layer in reality and I think that's the point of confusion.  To me, it looks like we are really just in a race to see if the JVM goes down before the request makes it out of the socket buffer inside the JVM.  Have I misunderstood the semantics?